modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/api.ts

```diff
@@ -72,9 +74,17 @@ export async function getFields(fdsView: FDSViewType) {
 			return;
 		}
 
+		if (propertyKey.includes(localizablePropertySuffix)) {
+			return;
+		}
+
+		if (propertyValue.additionalProperties) {
+			return;
+		}
```

Comment:
Yes, I was trying to "block" non field properties taking a look at the object api. I will change to be more strict, allowing only desired properties.

Reply:
For Object resources, I believe we can use `extensions` property as criteria, as I mentioned in https://github.com/liferay-frontend/liferay-portal/pull/3543#discussion_r1276383400. For other headless resources, I'm not sure if whitelisting or blacklisting is better :thinking:

Reply:
Just updated with `extensions'  and `{x-parent-map: properties}` for Object resources.

Reply:
> I'm not sure if whitelisting or blacklisting is better

This is a bit of a product question, but I think blacklisting is better:
- The worst thing that can happen with blacklisting is that some new or unpredicted meta field is added to the list of fields. It's annoying, but admin can ignore this, until we eventually patch it up.
- The worst thing that can happen with whitelisting is new regular field, with an unexpected definition format, does not show in the list. This could be a critical issue for a customer. The default cell renderer in FDS may not be able to handle the unexpected format, but there is a way around it: a client extension cell renderer.

Reply:
In other words, I don't think that allowing only desired properties is the right approach. Instead, if it's possible, it probably a better idea to allow everything, and ignore only a fixed set of properties.

Reply:
I think that for this blacklist/whitelist there is a [spike](https://liferay.atlassian.net/browse/LPS-191072)

Reply:
I left a comment in the spike, linking to here.